<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1445</width>
    <height>1169</height>
   </rect>
  </property>
  <property name="sizePolicy">
   <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
    <horstretch>0</horstretch>
    <verstretch>0</verstretch>
   </sizepolicy>
  </property>
  <property name="focusPolicy">
   <enum>Qt::StrongFocus</enum>
  </property>
  <property name="windowTitle">
   <string>Puzzle</string>
  </property>
  <property name="windowIcon">
   <iconset>
    <normaloff>icons/components.png</normaloff>icons/components.png</iconset>
  </property>
  <widget class="QWidget" name="centralWidget">
   <property name="sizePolicy">
    <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
     <horstretch>0</horstretch>
     <verstretch>0</verstretch>
    </sizepolicy>
   </property>
   <layout class="QHBoxLayout" name="horizontalLayout">
    <item>
     <widget class="PuzzleView" name="puzzleWidget" native="true">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
        <horstretch>1</horstretch>
        <verstretch>1</verstretch>
       </sizepolicy>
      </property>
      <property name="minimumSize">
       <size>
        <width>0</width>
        <height>0</height>
       </size>
      </property>
     </widget>
    </item>
    <item>
     <spacer name="horizontalSpacer">
      <property name="orientation">
       <enum>Qt::Horizontal</enum>
      </property>
      <property name="sizeHint" stdset="0">
       <size>
        <width>40</width>
        <height>20</height>
       </size>
      </property>
     </spacer>
    </item>
    <item>
     <widget class="PiecesView" name="piecesWidget" native="true">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
        <horstretch>1</horstretch>
        <verstretch>1</verstretch>
       </sizepolicy>
      </property>
      <property name="minimumSize">
       <size>
        <width>0</width>
        <height>0</height>
       </size>
      </property>
     </widget>
    </item>
    <item>
     <layout class="QVBoxLayout" name="verticalLayout">
      <item>
       <widget class="QPushButton" name="newGameButton">
        <property name="font">
         <font>
          <pointsize>10</pointsize>
         </font>
        </property>
        <property name="text">
         <string>New Game</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="sizeButton">
        <property name="font">
         <font>
          <pointsize>10</pointsize>
         </font>
        </property>
        <property name="text">
         <string>Size</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="closeButton">
        <property name="font">
         <font>
          <pointsize>10</pointsize>
         </font>
        </property>
        <property name="text">
         <string>Close</string>
        </property>
       </widget>
      </item>
      <item>
       <spacer name="verticalSpacer_2">
        <property name="orientation">
         <enum>Qt::Vertical</enum>
        </property>
        <property name="sizeType">
         <enum>QSizePolicy::Fixed</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>20</width>
          <height>40</height>
         </size>
        </property>
       </spacer>
      </item>
      <item>
       <widget class="QPushButton" name="saveButton">
        <property name="font">
         <font>
          <pointsize>10</pointsize>
         </font>
        </property>
        <property name="text">
         <string>Save</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="openButton">
        <property name="font">
         <font>
          <pointsize>10</pointsize>
         </font>
        </property>
        <property name="text">
         <string>Open</string>
        </property>
       </widget>
      </item>
      <item>
       <spacer name="verticalSpacer_3">
        <property name="orientation">
         <enum>Qt::Vertical</enum>
        </property>
        <property name="sizeType">
         <enum>QSizePolicy::Fixed</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>20</width>
          <height>40</height>
         </size>
        </property>
       </spacer>
      </item>
      <item>
       <widget class="QPushButton" name="rulesButton">
        <property name="font">
         <font>
          <pointsize>10</pointsize>
         </font>
        </property>
        <property name="text">
         <string>Rules</string>
        </property>
       </widget>
      </item>
      <item>
       <spacer name="verticalSpacer_4">
        <property name="orientation">
         <enum>Qt::Vertical</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>20</width>
          <height>40</height>
         </size>
        </property>
       </spacer>
      </item>
      <item>
       <widget class="QLabel" name="scoreLTextLabel">
        <property name="font">
         <font>
          <pointsize>14</pointsize>
         </font>
        </property>
        <property name="text">
         <string>Pieces left:</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLabel" name="scoreLNumLabel">
        <property name="font">
         <font>
          <pointsize>28</pointsize>
          <weight>75</weight>
          <bold>true</bold>
         </font>
        </property>
        <property name="text">
         <string/>
        </property>
       </widget>
      </item>
      <item>
       <spacer name="verticalSpacer">
        <property name="orientation">
         <enum>Qt::Vertical</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>20</width>
          <height>40</height>
         </size>
        </property>
       </spacer>
      </item>
      <item>
       <widget class="QPushButton" name="aboutButton">
        <property name="font">
         <font>
          <pointsize>10</pointsize>
         </font>
        </property>
        <property name="text">
         <string>About</string>
        </property>
       </widget>
      </item>
      <item>
       <spacer name="verticalSpacer_5">
        <property name="orientation">
         <enum>Qt::Vertical</enum>
        </property>
        <property name="sizeType">
         <enum>QSizePolicy::Fixed</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>77</width>
          <height>37</height>
         </size>
        </property>
       </spacer>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>PuzzleView</class>
   <extends>QWidget</extends>
   <header>puzzleview.h</header>
   <container>1</container>
   <slots>
    <signal>placeIsSelectedSignal(int)</signal>
    <signal>pieceToMoveSignal(int,int)</signal>
    <signal>pieceToDeleteSignal(int)</signal>
    <signal>pieceToRightPlaceSignal(int)</signal>
   </slots>
  </customwidget>
  <customwidget>
   <class>PiecesView</class>
   <extends>QWidget</extends>
   <header>piecesview.h</header>
   <container>1</container>
   <slots>
    <signal>pieceIsSelectedSignal(QImage)</signal>
   </slots>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections>
  <connection>
   <sender>puzzleWidget</sender>
   <signal>placeIsSelectedSignal(int)</signal>
   <receiver>MainWindow</receiver>
   <slot>placeIsSelectedSlot(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>267</x>
     <y>230</y>
    </hint>
    <hint type="destinationlabel">
     <x>535</x>
     <y>259</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>piecesWidget</sender>
   <signal>pieceIsSelectedSignal(QImage)</signal>
   <receiver>MainWindow</receiver>
   <slot>pieceIsSelectedSlot(QImage)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>1250</x>
     <y>212</y>
    </hint>
    <hint type="destinationlabel">
     <x>545</x>
     <y>199</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>puzzleWidget</sender>
   <signal>pieceToMoveSignal(int,int)</signal>
   <receiver>MainWindow</receiver>
   <slot>pieceToMoveSlot(int,int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>470</x>
     <y>391</y>
    </hint>
    <hint type="destinationlabel">
     <x>529</x>
     <y>394</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>puzzleWidget</sender>
   <signal>pieceToDeleteSignal(int)</signal>
   <receiver>MainWindow</receiver>
   <slot>pieceToDeleteSlot(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>568</x>
     <y>526</y>
    </hint>
    <hint type="destinationlabel">
     <x>620</x>
     <y>536</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>puzzleWidget</sender>
   <signal>pieceToRightPlaceSignal(int)</signal>
   <receiver>MainWindow</receiver>
   <slot>pieceToRightPlaceSlot(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>561</x>
     <y>469</y>
    </hint>
    <hint type="destinationlabel">
     <x>617</x>
     <y>471</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>placeIsSelectedSlot(int)</slot>
  <slot>pieceIsSelectedSlot(QImage)</slot>
  <slot>pieceToMoveSlot(int,int)</slot>
  <slot>pieceToDeleteSlot(int)</slot>
  <slot>pieceToRightPlaceSlot(int)</slot>
 </slots>
</ui>
